components/feature/tabs/src/main/java/mozilla/components/feature/tabs/WindowFeature.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.tabs
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.window.WindowRequest
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+
+/**
+ * Feature implementation for handling window requests by opening and closing tabs.
+ */
+class WindowFeature(
+    private val store: BrowserStore,
+    private val tabsUseCases: TabsUseCases
+) : LifecycleAwareFeature {
+
+    private var scope: CoroutineScope? = null
+
+    /**
+     * Starts observing the selected session to listen for window requests
+     * and opens / closes tabs as needed.
+     */
+    override fun start() {
+        scope = store.flowScoped { flow ->
+            flow.mapNotNull { state -> state.selectedTab }
```

Comment:
I think all of my questions above apply here as well, since the same underlying primitives are used.

Reply:
Yes, same pattern, see above.